src/values.js

```diff
@@ -160,10 +183,16 @@ function SetRef(value) {
 util.inherits(SetRef, Value);
 
 /** @ignore */
-SetRef.prototype.inspect = function() {
-  return 'SetRef(' + JSON.stringify(this.value) + ')';
+SetRef.prototype.toString = function() {
+  return 'SetRef(' + stringify(this.value) + ')';
 };
 
+SetRef.prototype.inspect = SetRef.prototype.toString;
+
+if (customInspect) {
```

Comment:
Maybe you could wrap everything in a function like `wrapToString(Ref, function() { ... }` just so we don't forget to always add `toString`, `inspect, and `customInspect`?